R/coord-.r

```diff
@@ -64,10 +64,10 @@ Coord <- ggproto("Coord",
   render_fg = function(panel_params, theme) element_render(theme, "panel.border"),
 
   render_bg = function(panel_params, theme) {
-    x.major <- if (length(panel_params$x.major) > 0) unit(panel_params$x.major, "native")
-    x.minor <- if (length(panel_params$x.minor) > 0) unit(panel_params$x.minor, "native")
-    y.major <- if (length(panel_params$y.major) > 0) unit(panel_params$y.major, "native")
-    y.minor <- if (length(panel_params$y.minor) > 0) unit(panel_params$y.minor, "native")
+    x.major <- if (length(panel_params$x.major) > 0) panel_params$x.major
+    x.minor <- if (length(panel_params$x.minor) > 0) panel_params$x.minor
+    y.major <- if (length(panel_params$y.major) > 0) panel_params$y.major
+    y.minor <- if (length(panel_params$y.minor) > 0) panel_params$y.minor
 
```

Comment:
It's not clear to me whether `guide_grid()` takes numerical values or units as arguments. If I understand correctly, it hands these arguments off to `element_render()` which ultimately hands them of to `polylineGrob()` which can take either numeric values or units for its x and y arguments. I think this ambiguity in what arguments grobs take leads to a lot of confusing code and potential errors.

Maybe, as a small step forward, it would help to add a comment to the definition of `guide_grid()` explaining what the variable type and format of `x.minor`, `x.major`, etc should be.

Reply:
Yeah, some documentation is probably in order. The offending lines are the two calls to setdiff which removes any attributes on the output so that polylineGrob ends up constructing the unit with default.unit anyway